internal/cmd/api/server.go

```diff
@@ -0,0 +1,41 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	ihttp "card-match/internal/pkg/http"
+)
+
+type server struct {
+	applicationServer *http.Server
+}
+
+func NewServer() *server {
+	applicationServer := ihttp.NewApplicationServer(8080)
+
+	return &server{
+		applicationServer: applicationServer,
+	}
+
+}
+
+func (s *server) StartApplicationServer() {
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-quit
+		fmt.Print("Gracefully shutting down application server\n")
+		s.applicationServer.RegisterOnShutdown(func() {
```

Comment:
Seems like the shutdown sequence is a strange place to register a shutdown callback

Reply:
Moved it to the application server handler.